liquidator/liquidator.js

```diff
@@ -248,22 +268,105 @@ class Liquidator {
       );
 
       // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
-      try {
-        await liquidation.call({ from: this.account });
-      } catch (error) {
-        this.logger.error({
+      const notEnoughCollateral = await liquidation
+        .call({ from: this.account })
+        .then(() => false)
+        .catch(() => true);
+
+      // Attempts to swap some capital for tokenCurrency
+      // Mutates the state of notEnoughCollateral in this logical scope
+      if (notEnoughCollateral) {
+        const tokenCurrency = await this.empClient.getTokenCurrency();
+
+        this.logger.info({
           at: "Liquidator",
-          message:
-            "Failed to liquidate position: not enough synthetic (or large enough approval) to initiate liquidation❌",
-          sponsor: position.sponsor,
-          inputPrice: scaledPrice.toString(),
-          position: position,
-          minLiquidationPrice: this.liquidationMinPrice,
-          maxLiquidationPrice: maxCollateralPerToken.toString(),
-          tokensToLiquidate: tokensToLiquidate.toString(),
-          error
+          message: `Attempting to convert reserve currency ${this.reserveCurrencyAddress} to tokenCurrency ${tokenCurrency.address} ❌`
         });
-        continue;
+
+        // Reverse calculation to estimate how much capital we need to get the amount of `tokensToLiquidate`
+        const reserveWeiNeeded = await this.oneInch.getExpectedReturn({
+          fromToken: tokenCurrency.address,
+          toToken: this.reserveCurrencyAddress,
+          amountWei: tokensToLiquidate.toString()
+        });
+
+        // This is used as a reference point to determine slippage
+        const oneGweiReturn = await this.oneInch.getExpectedReturn({
```

Comment:
Is there any reason not to parallelize the three async function calls (`.getTokenCurrency()`, `getExpectedReturn()`, `getExpectedReturn()`)?